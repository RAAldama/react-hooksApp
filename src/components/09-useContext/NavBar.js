import React from "react";
import { Link, NavLink } from "react-router-dom";

const NavBar = () => {
  return (
    <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
      <div className="container-fluid">

        <Link to="/" className="navbar-brand">HooksApp!</Link>
        
        <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
          <div className="navbar-nav">
            <NavLink exact activeClassName="active" to="/" className="nav-link">Home</NavLink>
            <NavLink exact activeClassName="active" to="/about" className="nav-link">About</NavLink>
            <NavLink exact activeClassName="active" to="/login" className="nav-link">Login</NavLink>
            <NavLink exact activeClassName="active" to="/other" className="nav-link">React Hooks</NavLink>
            <NavLink exact activeClassName="active" to="/todo" className="nav-link">useReducer</NavLink>
          </div>
        </div>

      </div>
    </nav>
  );
};

export default NavBar;
