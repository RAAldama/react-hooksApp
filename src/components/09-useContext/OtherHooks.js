import React from "react";
import CounterApp from "../01-useState/CounterApp";
import CounterCustom from "../01-useState/CounterCustom";
import FormCustom from "../02-useEffect/FormCustom";
import SimpleForm from "../02-useEffect/SimpleForm";
import MultipleHooks from "../03-examples/MultipleHooks";
import FocusScreen from "../04-useRef/FocusScreen";
import RealRef from "../04-useRef/RealRef";
import Layout from "../05-useLayoutEffect/Layout";
import CallbackHook from "../06-memos/CallbackHook";
import MemoHook from "../06-memos/MemoHook";
import Memorize from "../06-memos/Memorize";
import { Padre } from "../07-tarea-memo/Padre";

const OtherHooks = () => {
  return (
    <>
      <CounterApp />
      <hr />
      <CounterCustom />
      <hr />
      <SimpleForm />
      <hr />
      <FormCustom />
      <hr />
      <MultipleHooks />
      <hr />
      <FocusScreen />
      <hr />
      <RealRef />
      <hr />
      <Layout />
      <hr />
      <Memorize />
      <hr />
      <MemoHook />
      <hr />
      <CallbackHook />
      <hr />
      <Padre />
    </>
  );
};

export default OtherHooks;
