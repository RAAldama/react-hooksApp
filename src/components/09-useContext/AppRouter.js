import React from 'react';
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Redirect,
} from "react-router-dom";

import AboutScreen from './AboutScreen';
import LoginScreen from './LoginScreen';
import HomeScreen from './HomeScreen';
import NavBar from './NavBar';
import OtherHooks from './OtherHooks';
import TodoApp from '../08-useReducer/TodoApp';

const AppRouter = () => {
    return (
        <Router>
            <div>
                <NavBar />

                <div className="container">
                    <Switch>
                        <Route exact path="/about" component={ AboutScreen } />
                        <Route exact path="/login" component={ LoginScreen } />
                        <Route exact path="/other" component={ OtherHooks } />
                        <Route exact path="/todo" component={ TodoApp } />

                        {/* Usualmente poner la ruta / hasta el final */}
                        <Route exact path="/" component={ HomeScreen } />
                        <Redirect to="/" />
                    </Switch>
                </div>

            </div>
        </Router>
    )
}

export default AppRouter
