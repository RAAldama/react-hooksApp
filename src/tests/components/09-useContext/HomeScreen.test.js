import { mount } from "enzyme"
import HomeScreen from "../../../components/09-useContext/HomeScreen"
import { UserContext } from "../../../components/09-useContext/UserContext";

describe('tests on <HomeScreen />', () => {
    const user = {
        name: 'Albus',
        email: 'lol@lol.com'
    }

    const wrapper = mount(
        <UserContext.Provider value={{user}}>
            <HomeScreen />
        </UserContext.Provider>
    );
    
    test('should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })
    
})
