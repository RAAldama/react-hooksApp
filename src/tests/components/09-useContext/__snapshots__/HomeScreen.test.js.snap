// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on <HomeScreen /> should render correctly 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre>
    {
   "name": "Albus",
   "email": "lol@lol.com"
}
  </pre>
</div>
`;
